src/Catalyst.Core.Modules.Dfs/BlockExchange/Protocols/Bitswap11.cs

```diff
@@ -0,0 +1,271 @@
+
+        /// <inheritdoc />
+        public SemVersion Version { get; } = new SemVersion(1, 1);
+
+        /// <inheritdoc />
+        public override string ToString() { return $"/{Name}/{Version}"; }
+
+        /// <summary>
+        ///   The <see cref="BitswapService"/> service.
+        /// </summary>
+        public IBitswapService BitswapService { get; set; }
+
+        /// <inheritdoc />
+        public async Task ProcessMessageAsync(PeerConnection connection,
+            Stream stream,
+            CancellationToken cancel = default(CancellationToken))
+        {
+            // There is a race condition between getting the remote identity and
+            // the remote sending the first wantlist.
+            await connection.IdentityEstablished.Task.ConfigureAwait(false);
+
+            while (true)
+            {
+                var request = await ProtoBufHelper.ReadMessageAsync<Message>(stream, cancel).ConfigureAwait(false);
+
+                // Process want list
+                if (request.wantlist != null && request.wantlist.entries != null)
+                {
+                    foreach (var entry in request.wantlist.entries)
+                    {
+                        var cid = Cid.Read(entry.block);
+                        if (entry.cancel)
+                        {
+                            // TODO: Unwant specific to remote peer
+                            BitswapService.Unwant(cid);
+                        }
+                        else
+                        {
+                            // TODO: Should we have a timeout?
+                            var _ = GetBlockAsync(cid, connection.RemotePeer, CancellationToken.None);
+                        }
+                    }
+                }
+
+                // Forward sent blocks to the block service.  Eventually
+                // bitswap will here about and them and then continue
+                // any tasks (GetBlockAsync) waiting for the block.
+                if (request.payload != null)
+                {
+                    log.Debug($"got block(s) from {connection.RemotePeer}");
+                    foreach (var sentBlock in request.payload)
+                    {
+                        using (var ms = new MemoryStream(sentBlock.prefix))
+                        {
+                            var version = ms.ReadVarint32();
+                            var contentType = ms.ReadMultiCodec().Name;
+                            var multiHash = MultiHash.GetHashAlgorithmName(ms.ReadVarint32());
+                            await BitswapService.OnBlockReceivedAsync(connection.RemotePeer, sentBlock.data, contentType,
+                                multiHash);
+                        }
+                    }
+                }
+            }
+        }
+
+        async Task GetBlockAsync(Cid cid, Peer remotePeer, CancellationToken cancel)
+        {
+            // TODO: Determine if we will fetch the block for the remote
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)